curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635085f80bd9b5429faa40c4.md

```diff
@@ -0,0 +1,147 @@
+---
+id: 635085f80bd9b5429faa40c4
+title: Step 15
+challengeType: 0
+dashedName: step-15
+---
+
+# --description--
+
+You can actually clean this logic up a bit. Using the implicit return of an arrow function, you can directly return the value of the `.reduce()` method divided by the length of the array, without having to assign any variables.
+
+Update your `getMean` function as described above.
+
+# --hints--
+
+You should remove the `sum` variable declaration.
+
+```js
+assert.notMatch(getMean.toString(), /sum\s*=/);
+```
+
+You should remove the `mean` variable declaration.
+
+```js
+assert.notMatch(getMean.toString(), /mean\s*=/);
+```
+
+You should not change the logic within the `reduce` method.
+
+```js
+assert.match(getMean.toString(), /(array\.reduce\(\(acc\s*,\s*el\s*\)\s*=>|array\.reduce\(function\s*\(acc\s*,\s*el\)\s*\{)\s*(return)?\s*acc\s*\+\s*el;\s*\}?\s*,\s*0\)/)
+```
+
+You should divide the result of the `reduce` method by the length of the array.
+
+```js
+assert.match(getMean.toString(), /(array\.reduce\(\(acc\s*,\s*el\s*\)\s*=>|array\.reduce\(function\s*\(acc\s*,\s*el\)\s*\{)\s*(return)?\s*acc\s*\+\s*el;\s*\}?\s*,\s*0\)\s*\/\s*array\.length/)
+```
+
+You should use implicit return syntax to directly return the result of `reduce` divided by the array length.
+
+```js
+assert.match(code, /const\s+getMean\s*=\s*\(?array\)?\s*=>\s*array\.reduce\(\(acc\s*,\s*el\)\s*=>\s*acc\s*\+\s*el\s*,\s*0\s*\)\s*\/\s*array\.length;/)
```

Comment:
The tests will fail if the camper does not add a semicolon at the end of their answer.
I don't think we should be that strict with semicolons and allow answers without the semicolon to pass. 

<img width="884" alt="step-15-failing-test-missing-semicolon" src="https://user-images.githubusercontent.com/67210629/211250149-bd3b068e-6002-4a7d-86ea-2a5faeed7f2c.png">